adapters/redis/test/opentelemetry/adapters/redis/adapter_test.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../lib/opentelemetry/adapters/redis'
+require_relative '../../../../lib/opentelemetry/adapters/redis/patches/client'
+
+describe OpenTelemetry::Adapters::Redis::Adapter do
+  let(:adapter) { OpenTelemetry::Adapters::Redis::Adapter.instance }
+  let(:exporter) { EXPORTER }
+  let(:span) { exporter.finished_spans.first }
+
+  before do
+    exporter.reset
+  end
+
+  after do
+    # Force re-install of instrumentation
+    adapter.instance_variable_set(:@installed, false)
+  end
+
+  describe 'tracing' do
+    before do
+      adapter.install
+    end
+
+    it 'before request' do
+      _(exporter.finished_spans.size).must_equal 0
+    end
+
+    it 'after request' do
+      ::Redis.new.set('mykey', 'hello world')
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'SET'
+      _(span.attributes['component']).must_equal 'redis'
+      _(span.attributes['db.type']).must_equal 'redis'
+      _(span.attributes['db.instance']).must_equal '0'
+      _(span.attributes['db.statement']).must_equal 'SET mykey hello world'
+      _(span.attributes['db.url']).must_equal 'redis://127.0.0.1:6379'
+      _(span.attributes['net.peer.name']).must_equal '127.0.0.1'
+      _(span.attributes['net.peer.port']).must_equal 6379
+    end
+
+    it 'after pipeline' do
+      ::Redis.new(host: 'example.com', port: '8321').pipelined do |redis|
+        redis.set('v1', '0')
+        redis.incr('v1')
+        redis.get('v1')
+      end
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'pipeline'
+      _(span.attributes['component']).must_equal 'redis'
+      _(span.attributes['db.type']).must_equal 'redis'
+      _(span.attributes['db.instance']).must_equal '0'
+      _(span.attributes['db.statement']).must_equal "SET v1 0\nINCR v1\nGET v1"
```

Comment:
We will eventually have a array valued attributes. This might be a good use for them, but for now, this works.